releases/release-5.1.2.md

```diff
@@ -0,0 +1,156 @@
+---
+title: TiDB 5.1.2 Release Notes
+---
+
+# TiDB 5.1.2 Release Notes
+
+Release Date: September 27, 2021
+
+TiDB version: 5.1.2
+
+## Improvements
+
++ TiDB
+
+    - Trigger auto-analyze based on histogram row count [#26708](https://github.com/pingcap/tidb/pull/26708)
+
++ TiFlash
+
+    - Support pushing down ROUND(X) to TiFlash. [#2611](https://github.com/pingcap/tics/pull/2611)
+    - support function DATE() pushed down to tiflash [#2602](https://github.com/pingcap/tics/pull/2602)
+    - support functions of ADDDATE() and DATE_ADD() pushed down to tiflash [#2591](https://github.com/pingcap/tics/pull/2591)
+    - retry when meeting stablish conn fails [#2570](https://github.com/pingcap/tics/pull/2570)
+    - Support pushing down MOD(N, M) to TiFlash. [#2560](https://github.com/pingcap/tics/pull/2560)
+    - Add Grafana panels for write throughput per instance [#2544](https://github.com/pingcap/tics/pull/2544)
+    -  Cherry-pick #2216 #2460 #2494 & Optimizr read-index retry to release-5.1 [#2345](https://github.com/pingcap/tics/pull/2345)
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix wrong index hash join when hash col is enum. [#28081](https://github.com/pingcap/tidb/pull/28081)
+    - Fix a batch client bug that recycle idle connection may block sending requests in some rare cases. [#27678](https://github.com/pingcap/tidb/pull/27678)
+    - Keep the overflow check logic same with mysql [#26725](https://github.com/pingcap/tidb/pull/26725)
+    - Fix #26147 that `tidb` returns an `unknow` error with no message while it should return the error that contains `pd is timeout`. [#26682](https://github.com/pingcap/tidb/pull/26682)
+    - Fix wrong charset and collation for case when function [#26673](https://github.com/pingcap/tidb/pull/26673)
+    - Fix the issue that greatest(datetime) union null returns empty string [#26566](https://github.com/pingcap/tidb/pull/26566)
+
```

Comment:
```suggestion

+ TiKV

    - Fix bug when TiKV upgrade from 3.0 to 4.x and 5.x but there are still some files left by import. [#10912](https://github.com/tikv/tikv/pull/10912)
    - RaftStore Snapshot GC fix: fix the issue that snapshot GC missed GC snapshot files when there's one snapshot file failed to be GC-ed. [#10873](https://github.com/tikv/tikv/pull/10873)
    - TiKV coprocessor slow log will only consider time spent on processing the request.
    - Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. [#10865](https://github.com/tikv/tikv/pull/10865)
    - Bug fix: fix an unexpected panic when exceeds deadline on processing copr requests. [#10856](https://github.com/tikv/tikv/pull/10856)
    - Fix TiKV panic when enable Titan and upgrade from pre-5.0 version
    - Fix newer TiKV can't rollback to 5.0.x [#10842](https://github.com/tikv/tikv/pull/10842)
    - Fix TiKV delete files before it ingests to RocksDB. [#10741](https://github.com/tikv/tikv/pull/10741)
    - Fix the resolve failures caused by the left pessimisic locks. [#10653](https://github.com/tikv/tikv/pull/10653)

```